packages/gen-ai/frontend/src/app/AIAssets/components/MaaSModelTableRow.tsx

```diff
@@ -1,59 +1,146 @@
+import { AIModel, LlamaModel, LlamaStackDistributionModel, MaaSModel } from '~/app/types';
+import { GenAiContext } from '~/app/context/GenAiContext';
+import { genAiChatPlaygroundRoute } from '~/app/utilities/routes';
+import ChatbotConfigurationModal from '~/app/Chatbot/components/chatbotConfiguration/ChatbotConfigurationModal';
 import MaaSModelTableRowEndpoint from './MaaSModelTableRowEndpoint';
 
 type MaaSModelTableRowProps = {
   model: MaaSModel;
   namespace: string;
+  playgroundModels: LlamaModel[];
+  lsdStatus: LlamaStackDistributionModel | null;
+  aiModels: AIModel[];
+  maasModels: MaaSModel[];
 };
 
-const MaaSModelTableRow: React.FC<MaaSModelTableRowProps> = ({ model, namespace }) => (
-  <Tr>
-    <Td dataLabel="Model deployment name">
-      <TableRowTitleDescription
-        title={
-          <>
-            {model.id}
-            <Popover aria-label="Models as a Service" bodyContent={<>Models as a Service</>}>
-              <Label
-                style={{ marginLeft: 'var(--pf-t--global--spacer--sm)' }}
-                color="orange"
-                aria-label="Model as a Service"
-              >
-                MaaS
-              </Label>
-            </Popover>
-          </>
-        }
-      />
-    </Td>
-    <Td dataLabel="External endpoint">
-      <MaaSModelTableRowEndpoint model={model} namespace={namespace} />
-    </Td>
-    <Td dataLabel="Status">
-      {model.ready ? (
-        <Label
-          color="green"
-          icon={<CheckCircleIcon aria-label="Active status" />}
-          aria-label={`${model.id} status: Active`}
-        >
-          Active
-        </Label>
-      ) : (
-        <Label
-          color="red"
-          icon={<ExclamationCircleIcon aria-label="Inactive status" />}
-          aria-label={`${model.id} status: Inactive`}
-        >
-          Inactive
-        </Label>
+const MaaSModelTableRow: React.FC<MaaSModelTableRowProps> = ({
+  model,
+  namespace,
+  playgroundModels,
+  lsdStatus,
+  aiModels,
+  maasModels,
+}) => {
+  const navigate = useNavigate();
+  const { namespace: contextNamespace } = React.useContext(GenAiContext);
+  const enabledModel = playgroundModels.find((m) => m.modelId === model.id);
+  const [isConfigurationModalOpen, setIsConfigurationModalOpen] = React.useState(false);
+
+  // Convert MaaS model to AIModel format for pre-selection in the modal
+  const maasAsAIModel: AIModel = React.useMemo(
+    () => ({
+      model_name: model.id,
+      model_id: model.id,
+      serving_runtime: 'MaaS',
+      api_protocol: 'OpenAI',
+      version: '',
+      usecase: 'LLM',
+      description: `Model as a Service - ${model.owned_by}`,
+      endpoints: [`internal: ${model.url}`],
+      status: 'Running' as const,
+      display_name: model.id,
+      sa_token: {
+        name: '',
+        token_name: '',
+        token: '',
+      },
+      internalEndpoint: model.url,
+      isMaaSModel: true,
+      maasModelId: model.id,
+    }),
+    [model],
```

Comment:
Can we make this as a util function? I see it's also used in the configuration modal.